webapp/src/components/modals/configModal.tsx

```diff
@@ -0,0 +1,760 @@
+    const [actionChannelsAddedToValid, setActionChannelsAddedToValid] = useState(false);
+    const [actionSuccessfullMessageValid, setActionSuccessfullMessageValid] = useState(false);
+    const [actionNameValid, setActionNameValid] = useState(false);
+
+    const [validated, setValidated] = useState(false);
+
+    const [deleteAction, setDeleteAction] = useState('');
+
+    const [selectedTeam, setSelectedTeam] = useState('');
+
+    const [teamSelectionWarning, setTeamSelectionWarning] = useState(false);
+
+    const [actionClicked, setActionClicked] = useState(false);
+
+    const [teamOptionList, setTeamOptionList] = useState<GroupTypes[]>([]);
+    const [channelOptionList, setChannelOptionList] = useState<OptionTypes[]>([]);
+
+    const [teamDropdownDisabled, setTeamDropdownDisabled] = useState(false);
+    const [teamApiCalled, setTeamApiCalled] = useState(true);
+    const [teamApiError, setTeamApiError] = useState('');
+
+    const [chanelDropdownDisabled, setChannelDropdownDisabled] = useState(false);
+    const [channelApiCalled, setChannelApiCalled] = useState(true);
+    const [channelApiError, setChannelApiError] = useState('');
+
+    const actionLength = existingConfig?.actions?.length ?? 0;
+
+    const reduxState = useSelector((state: GlobalState) => state);
+    const mmSiteUrl = reduxState?.entities?.general?.config?.SiteURL as string;
+
+    useEffect(() => {
+        setShow(visibility);
+        setIsConfigVisible(visibility);
+    }, [visibility]);
+
+    useEffect(() => {
+        setExistingConfig(configIndex === null ? newConfig : config[configIndex]);
+    }, [config]);
+
+    useEffect(() => {
+        getTeam(mmSiteUrl);
+        getChannel(mmSiteUrl);
+        if (configIndex !== null) {
+            setSelectedTeam(existingConfig.teamName);
+            setTeamName(existingConfig.teamName);
+            setDelay(existingConfig.delayInSeconds);
+            setMessage(existingConfig.message);
+            setGuestValue(existingConfig?.includeGuests ?? '');
+            setAttachmentMessage(existingConfig?.attachmentMessage ?? []);
+        }
+    }, []);
+
+    useEffect(() => {
+        preFillActions();
+    }, [actionIndex]);
+
+    useEffect(() => {
+        setTeamNameValid(Boolean(teamName.trim()));
+
+        setTeamSelectionWarning(Boolean(teamName.trim()));
+
+        if (message.length) {
+            setMessageValid(Boolean(message[0].trim()));
+        } else {
+            setMessageValid(false);
+        }
+
+        setDelayValid(delay >= 0);
+
+        setActionTypesValueValid(Boolean(actionTypesValue));
+
+        setActionDisplayNameValid(Boolean(actionDisplayName.trim()));
+
+        setActionNameValid(Boolean(actionName.trim()));
+
+        if (actionChannelsAddedTo.length) {
+            setActionChannelsAddedToValid(Boolean(actionChannelsAddedTo[0]));
+        } else {
+            setActionChannelsAddedToValid(false);
+        }
+
```

Comment:
```suggestion
        setActionChannelsAddedToValid(actionChannelsAddedTo.length ? Boolean(actionChannelsAddedTo[0]) : false);
```
Check if this works. If yes, fix everywhere

Reply:
no this will not in the case of arrays first element being a blank string, where I need a false but this code will give true

Reply:
I don't think so. Please re-verify. Also, please ask @avas27JTG if this is the correct method or not.

Reply:
Discussed with @avas27JTG, he suggested to keep the changes